erdos/cluster.py

```diff
@@ -0,0 +1,82 @@
+from pexpect import pxssh
+
+
+class Cluster(object):
+    """Cluster base class.
+
+    All clusters must inherent from this class, and must implement:
+
+    1. setup_node: Runs commands to connect the node to the cluster.
+    """
+
+    def __init__(self):
+        self.nodes = {}
+
+    def __del__(self):
+        for node in self.nodes.values():
+            node.disconnect()
+
+    def add_node(self,
```

Comment:
IMHO, we should plug out the node construction from `add_node`. Instead the method should receive a node object. In this way, we avoid coupling nodes with clusters, we can enrich nodes with more info (e.g., resources), and use node objects to do scheduling for ROS.

Reply:
Sounds good! I was struggling with this, and decided to implement a top-down approach, where nodes are the same and specialized clusters invoke different commands on nodes.

If we're looking for more fine-grained control on individual nodes, enriching nodes makes sense.